services/security/security.go

```diff
@@ -160,12 +160,13 @@ const (
 	StoragePermissionsUpdate StoragePermissions = "update"
 )
 
-type ClientType string
+type Operation string
 
 const (
-	ControlPlane   ClientType = "ControlPlane"
-	ExternalClient ClientType = "ExternalClient"
-	Node           ClientType = "Node"
+	ReadAccess   Operation = "read"
+	WriteAccess  Operation = "write"
```

Comment:
How do I differentiate between Creating a Group Vs Creating Resource in a Group.
What access should be available on Parent to allow this ie 
for group create, what permission in Location is required.

Reply:
For creating a group, a role with a permission containing write/all and a providerType of any or group would allow creation of a group. Assigning this role using a scope under a location (or not bound by any scope), will allow a client to create a group.
To create a resource in a group we can have a role with permissions to create the resource, and assign it using a scope that is specifying a specific group.

Reply:
Definitely need a huge md for this. Please try to get a md file after enabling the feature.